### PR TITLE
Fix ENV variable setting in JRF domain in PV test class

### PR DESCRIPTION
**Pull request by maggiehe00**
This PR is to set env variables in JRF domain in PV by grabbing the variables from docker image.

Test result using image: phx.ocir.io/weblogick8s/stage/fmw-infrastructure:12.2.1.4
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2517/testReport/
(<10-19-2020 22:36:28> <INFO> <oracle.weblogic.kubernetes.ItJrfDomainInPV testJrfDomainInPvUsingWlst> <JAVA_HOME in image localhost:5000/weblogick8s/stage/fmw-infrastructure:12.2.1.4 is: /usr/java/jdk-8>)

Test result using image: phx.ocir.io/weblogick8s/test-images/fmw-infrastructure:12.2.1.4
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2545/testReport/
(<JAVA_HOME in image localhost:5000/weblogick8s/test-images/fmw-infrastructure:12.2.1.4 is: /u01/jdk>)

After addressing the review comments, Jenkins run passed at
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2563/testReport/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Docker.java [hunk 1]
```diff
@@ -121,4 +125,26 @@ public static JsonObject createDockerConfigJson(String username, String password
...
+  public static String getImageEnvVar(String imageName, String envVarName) {
+    LoggingFacade logger = getLogger();
+    String cmdToExecute = String.format("docker run %s /bin/bash -c 'echo \"$%s\"'", imageName, envVarName);
+    logger.info("getImageEnvVar with imageName: {0}, envVarName: {1}", imageName, envVarName);
+    ExecResult result = Command.withParams(
+                              new CommandParams()
+                                .command(cmdToExecute)
+                                .saveResults(true)
+                                .redirect(true)
+                            ).executeAndReturnResult();
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Docker.java:144]: Can you check if the command is executed successfully or not by asserting the exit code or not null return string from exec command

**Review by anpanigr — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Docker.java (hunk 1 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Docker.java:144]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 2]
```diff
@@ -173,9 +176,16 @@ public void testJrfDomainInPvUsingWlst() {
     File domainPropertiesFile = assertDoesNotThrow(() ->
             File.createTempFile("domain", "properties"),
         "Failed to create domain properties file");
+
+    //get ENV variable from the image
+    oracle_home = getImageEnvVar(FMWINFRA_IMAGE_TO_USE_IN_SPEC, "ORACLE_HOME");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:181]: you can assert for null

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 3]
```diff
@@ -173,9 +176,16 @@ public void testJrfDomainInPvUsingWlst() {
...
+    logger.info("ORACLE_HOME in image {0} is: {1}", FMWINFRA_IMAGE_TO_USE_IN_SPEC, oracle_home);
+    java_home = getImageEnvVar(FMWINFRA_IMAGE_TO_USE_IN_SPEC, "JAVA_HOME");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:183]: add assert for null

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 2 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:183]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 3 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:187]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 4]
```diff
@@ -173,9 +176,20 @@ public void testJrfDomainInPvUsingWlst() {
     File domainPropertiesFile = assertDoesNotThrow(() ->
             File.createTempFile("domain", "properties"),
         "Failed to create domain properties file");
+
+    //get ENV variable from the image
+    assertNotNull(getImageEnvVar(FMWINFRA_IMAGE_TO_USE_IN_SPEC, "ORACLE_HOME"),
+        "envVar from image is null");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:182]:
```suggestion
        "envVar ORACLE_HOME from image is null");
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 5]
```diff
@@ -173,9 +176,20 @@ public void testJrfDomainInPvUsingWlst() {
...
+    oracle_home = getImageEnvVar(FMWINFRA_IMAGE_TO_USE_IN_SPEC, "ORACLE_HOME");
+    logger.info("ORACLE_HOME in image {0} is: {1}", FMWINFRA_IMAGE_TO_USE_IN_SPEC, oracle_home);
+    assertNotNull(getImageEnvVar(FMWINFRA_IMAGE_TO_USE_IN_SPEC, "JAVA_HOME"),
+        "envVar from image is null");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:186]:
```suggestion
        "envVar JAVA_HOME from image is null");
```

**Review by vanajamukkara — APPROVED** [review]
LG, include the var name in the log message.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 5 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java]: Done. Thanks!

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 4 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java]: Done. Thanks!

**Pull request merged**